File: backend/IMIS/IMIS.Domain/Announcement.cs
using Base.Primitives;

namespace IMIS.Domain
{
    public class Announcement : Entity<int>
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime FromDate { get; set; }
        public DateTime ToDate { get; set; }
        public bool IsActive { get; set; }
    }
}

File: frontend/imis/lib/announcements/models/announcement.dart
import 'package:json_annotation/json_annotation.dart';

part 'announcement.g.dart';

@JsonSerializable()
class Announcement {
  final int id;
  final String title;
  final String description;
  final DateTime fromDate;
  final DateTime toDate;
  final bool isActive;
  final String? rowVersion;
  final bool? isDeleted;

  Announcement({
    required this.id,
    required this.title,
    required this.description,
    required this.fromDate,
    required this.toDate,
    required this.isActive,
    this.rowVersion,
    this.isDeleted,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) =>
      _$AnnouncementFromJson(json);

  Map<String, dynamic> toJson() => _$AnnouncementToJson(this);
}
